src/EditorFeatures/CSharpTest/Completion/CompletionProviders/Snippets/CSharpConsoleSnippetCompletionProviderTests.cs

```diff
@@ -0,0 +1,363 @@
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInSwitchExpression()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+       var operation = 2;  
+  
+        var result = operation switch  
+        {
+            $$
+            1 => ""Case 1"",  
+            2 => ""Case 2"",  
+            3 => ""Case 3"",  
+            4 => ""Case 4"",  
+        };
+    }
+}";
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInSingleLambdaExpression()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+       Func<int, int> f = x => $$;
+    }
+}";
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInStringTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+        var str = ""$$"";
+    }
+}";
+
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInObjectInitializerTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+        var str = new Test($$);
+    }
+}
+
+class Test
+{
+    private string val;
+
+    public Test(string val)
+    {
+        this.val = val;
+    }
+}";
+
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInParameterListTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method(int x, $$)
+    {
+    }
+}";
+
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInRecordDeclarationTest()
+        {
+            var markupBeforeCommit =
+@"public record Person
+{
+    $$
+    public string FirstName { get; init; } = default!;
+    public string LastName { get; init; } = default!;
+};";
+
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task NoConsoleSnippetInVariableDeclarationTest()
+        {
+            var markupBeforeCommit =
+@"class Program
+{
+    public void Method()
+    {
+        var x = $$
+    }
+}";
+
+            await VerifyItemIsAbsentAsync(markupBeforeCommit, s_itemToCommit);
+        }
```

Comment:
do you have any tests where you've partially written the snippet?

Reply:
By that do you mean the invoking text for the completion item?

Reply:
yes.  e.g.  you've written `c`.